### DISCUSSION TITLE
Dynamic Sitemap

### DISCUSSION THREAD

**Discussion by goppi0** [vercel/vercel#6107] [Help] [answered]
Hi all, I'm new to Vercel and I try to use Vercel with a React website and for the time being without NextJS. The backend for my website updates daily a XML sitemap from a database and I need to find a way how i can offer the XML sitemap through the frontend. 

Vercel Support suggested to use a serveless API that could get the XML information from the backend and cache it for a day. The frontend would than "route" /sitemap.xml to the the API. They could however not give me any more detail or links to relevant documentation. Does anyone have an idea how to accomplish this or is Vercel not the right platform for this?

Thanks in advance,
Goppi

**Comment by fu-sen** [comment, marked as answer]
You run a Serverless Function that generates a sitemap.xml in `/api`:
https://vercel.com/docs/serverless-functions/introduction

If you want it to be visible in `/sitemap.xml`:
https://vercel.com/docs/configuration#project/rewrites

For example, **vercel.json:**
```
{
  "rewrites": [
    { "source": "/sitemap.xml", "destination": "/api/sitemap.js" },
  ]
}
```

Refer to the function directly instead of generating file `sitemap.xml`. Vercel has a cache feature that you can take advantage of:
https://vercel.com/docs/edge-network/caching

**Reply by goppi0** [reply]
Hi, thanks for the advice. Is a serverless function able to produce a gz file instead of the xml?
